packages/flutter/lib/src/material/tab_controller.dart

```diff
@@ -425,11 +429,16 @@ class _DefaultTabControllerState extends State<DefaultTabController> with Single
   @override
   void initState() {
     super.initState();
+    final int? initialIndex = PageStorage.of(context)?.readState(context) as int?;
     _controller = TabController(
       vsync: this,
       length: widget.length,
-      initialIndex: widget.initialIndex,
-    );
+      initialIndex: initialIndex ?? widget.initialIndex,
```

Comment:
Document on `initialIndex` that it is ignored if an index is found in PageStorage.